pkg/maps/ctmap/ctmap_mock.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ctmap
+
+import (
+	"github.com/cilium/cilium/pkg/bpf"
+)
+
+// A "Record" designates a map entry (key + value), but avoid "entry" because of
+// possible confusion with "CtEntry" (actually the value part).
+type CtMockMapRecord struct {
+	Key   CtKey
+	Value *CtEntry
+}
+
+// CtMockMap implements the CtMap interface and can be used for unit tests.
+type CtMockMap struct {
+	Entries []CtMockMapRecord
```

Comment:
Could you use `[]ctKeyVal` here? That would remove some code and simplify the `DeepEquals` `Assert()`.

Reply:
I can and it will remove one struct indeed, but the `Assert()` won't be any simpler. This is because we unmarshall JSON into `[]ctRecord4` and `[]ctRecord6`, which don't have a `ctmap.CtKey` as key (or unmarshalling would not work), so the `DeepEquals` find different objects.